AD_query_tool.py

```diff
@@ -1,75 +1,75 @@
-
 import os
 import ldap
 import csv
 import pandas as pd
 import datetime
 import dateutil.relativedelta
 
+LDAP_SERVER = 'ldap://172.20.1.216:389'
+BASE_DN = 'DC=lab, DC=visiblerisk, DC=com'
+LDAP_LOGIN = 'Administrator'
+LDAP_PASS = 'Nhwk2iISxJuOZQa43IDch0J8z0hqB5jY'
+SEARCH_OBJECT = '(&(objectClass=user)(objectCategory=person))'
+SEARCH_ATTR = ['userAccountControl', 'lastLogon', 'PwdLastSet', 'memberOf']
+CSV_PATH = os.path.join('.', 'AD_output.csv')  # path and file name for AD query results dataset  
 
-def active_directory_conn(ldap_server, ldap_login, ldap_pass):
-    conn = ldap.initialize(ldap_server)
-    conn.simple_bind_s(ldap_login, ldap_pass) 
+def active_directory_conn(LDAP_SERVER, LDAP_LOGIN, LDAP_PASS):
+    conn = ldap.initialize(LDAP_SERVER)
+    conn.simple_bind_s(LDAP_LOGIN, LDAP_PASS) 
     return(conn)
 
-def search_active_directory(base_dn, search_object,  search_attr, connection):
-    results = connection.search_s(base_dn, ldap.SCOPE_SUBTREE, search_object , search_attr)
+def search_active_directory(BASE_DN, search_object, search_attr, connection):
+    results = connection.search_s(BASE_DN, ldap.SCOPE_SUBTREE, search_object , search_attr)
     connection.unbind()
     return(results[:-4]) # cutting out irrelevant data rows 
 
-def results_to_csv(path, active_directory_query):
+def results_to_csv(path, active_directory_output):
     with open(path, 'w') as csv_file:  
         writer = csv.writer(csv_file)
         header = ['CN', 'GroupsMembership', 'UserAccountControl', 'LastLogon', 'LastPasswordSet'] 
         writer.writerow(header)
-        for i in active_directory_query:
+        for i in active_directory_output:
             cn = i[0]
             member_of = i[0],i[1].get('memberOf')
             user_account_control = int(i[1].get('userAccountControl')[0])
             last_logon = int(i[1].get('lastLogon')[0])
             pwd_last_set = int(i[1].get('pwdLastSet')[0])
             writer.writerow([cn, member_of ,user_account_control, last_logon, pwd_last_set])
 
-def enabled_users(df):    
-    enabled_users = df.loc[df['UserAccountControl'] & 2 == 0]
-    return(enabled_users)
-
-def enabled_users_amount():    
-    df = pd.read_csv(csv_path)
-    enabled_users_amount = enabled_users(df)['CN'].count()
-    return(enabled_users_amount)
-
+# This function used as reference datetime for calculating delta between datetimes
 def current_date_time():
     current_ts = datetime.datetime.now()
     current_date_time = current_ts.replace(microsecond=0)
     return(current_date_time)
 
+# This function converts the 18-digit Active Directory timestamp to human readable datetime
```

Comment:
https://www.python.org/dev/peps/pep-0257/ - this is how functions should be documented